File: server/models/Specialty.js
const mongoose = require('mongoose');

const specialtySchema = new mongoose.Schema({
  name: {
    type: String,
    required: [true, 'Tên chuyên khoa là bắt buộc'],
    unique: true,
    trim: true
  },
  description: {
    type: String,
    trim: true
  },
  icon: {
    type: String,
    trim: true
  },
  imageUrl: {
    type: String,
    trim: true
  },
  image: {
    url: String,
    secureUrl: String,
    publicId: String,
    cloudName: String
  },
  isActive: {
    type: Boolean,
    default: true
  }
}, {
  timestamps: true
});

const Specialty = mongoose.model('Specialty', specialtySchema);

module.exports = Specialty; 

File: hospital_mobile_app/lib/domain/entities/specialty.dart
import 'package:equatable/equatable.dart';

class Specialty extends Equatable {
  final String id;
  final String name;
  final String? description;
  final String? icon;
  final String? imageUrl;
  final int doctorCount;
  final int serviceCount;

  const Specialty({
    required this.id,
    required this.name,
    this.description,
    this.icon,
    this.imageUrl,
    required this.doctorCount,
    required this.serviceCount,
  });

  @override
  List<Object?> get props => [
        id,
        name,
        description,
        icon,
        imageUrl,
        doctorCount,
        serviceCount,
      ];
}
